fairing/notebook/jobs.py

```diff
@@ -0,0 +1,61 @@
+import fairing
+from fairing.functions.function_shim import get_execution_obj_type, ObjectType
+from fairing.preprocessors.function import FunctionPreProcessor
+from fairing.builders.docker.docker import DockerBuilder
+from fairing.deployers.job.job import Job
+from fairing.deployers.serving.serving import Serving
+
+def guess_preprocessor(obj, *args, **kwargs):
+    if get_execution_obj_type(obj) != ObjectType.NOT_SUPPORTED:
+        return FunctionPreProcessor(obj, *args, **kwargs) 
+    else:
+        raise NotImplementedError("obj param should be a function or a class, got {}".format(type(obj)))
+
+def guess_docker_registry():
+    try:
+        gcp_project = fairing.cloud.gcp.guess_project_name()
+        registry = 'gcr.io/{}'.format(gcp_project)
+        return registry
+    except Exception as e:
+        return None
+
+class BaseJob:
+    
+    def __init__(self, obj, base_docker_image, docker_registry=None, input_files=[]):
+        if not docker_registry:
+            guessed_docker_registry = guess_docker_registry()
+            if not guessed_docker_registry:
+                raise RuntimeError("Not able to guess docker registry, please specify docker_registry explicitly.")
+            self.docker_registry = guessed_docker_registry
+        else:
+            self.docker_registry = docker_registry
+        preprocessor = guess_preprocessor(obj, input_files=input_files)
+        self.builder = DockerBuilder(preprocessor=preprocessor,
```

Comment:
Suggest using the Cluster builder instead, which will work on Kubeflow notebooks

Reply:
Yes cluster builder makes sense and also useful to reduce network bandwidth if the user is developing out of their notebook.